src/Private/OpenApi.ps1

```diff
@@ -105,78 +186,165 @@ function Test-PodeOAComponentParameter {
+            if ($Property.minItems ) {
+                $schema['minItems'] = $Property.minItems
+            }
 
-    # are we using an object?
-    if ($Property.object) {
-        $Property.object = $false
+            if ($Property.uniqueItems ) {
+                $schema['uniqueItems'] = $Property.uniqueItems
+            }
 
-        $schema = @{
-            type       = 'object'
-            properties = (ConvertTo-PodeOASchemaObjectProperty -Properties $Property)
+            $schema['type'] = 'array'
+            if ($Property.type -ieq 'schema') {
+                if ( !(Test-PodeOAComponentSchema -Name $Property['schema'])) {
+                    throw "The OpenApi component schema doesn't exist: $($Property['schema'])"
+                }
+                $schema['items'] = @{ '$ref' = "#/components/schemas/$($Property['schema'])" }
+            } else {
+                $Property.array = $false
+                $schema['items'] = ($Property | ConvertTo-PodeOASchemaProperty)
+                $Property.array = $true
+            }
+            return $schema
+        } else {
+            #format is not applicable to array
+            if ($Property.format) {
+                $schema['format'] = $Property.format
+            }
+
+            # schema refs
+            if ($Property.type -ieq 'schema') {
+                if ( !(Test-PodeOAComponentSchema -Name $Property['schema'])) {
+                    throw "The OpenApi component schema doesn't exist: $($Property['schema'])"
+                }
+                $schema = @{
+                    '$ref' = "#/components/schemas/$($Property['schema'])"
+                }
+            }
+            #only if it's not an array
+            if ($Property.enum ) {
+                $schema['enum'] = $Property.enum
+            }
         }
 
-        if ($Property.required) {
-            $schema['required'] = @($Property.name)
+        if ($Property.object) {
+            # are we using an object?
+            $Property.object = $false
+
+            $schema = @{
+                type       = 'object'
+                properties = (ConvertTo-PodeOASchemaObjectProperty -Properties $Property)
+            }
+
+            if ($Property.required) {
+                $schema['required'] = @($Property.name)
+            }
         }
-    }
 
-    if ($Property.type -ieq 'object') {
-        $schema['properties'] = (ConvertTo-PodeOASchemaObjectProperty -Properties $Property.properties)
-        $schema['required'] = @(($Property.properties | Where-Object { $_.required }).name)
-    }
+        if ($Property.type -ieq 'object') {
+            foreach ($prop in $Property.properties) {
+                if ( @('allOf', 'oneOf', 'anyOf') -icontains $prop.type  ) {
+                    switch ($prop.type) {
+                        'allof' { $prop.type = 'allOf' }
+                        'oneof' { $prop.type = 'oneOf' }
+                        'anyof' { $prop.type = 'anyOf' }
+                    }
```

Comment:
To be safe: `switch ($prop.type.ToLower())`; this way it matches the if-statement just above it.